server/src/main/java/io/crate/analyze/RestoreSnapshotAnalyzer.java

```
@@ -61,16 +76,79 @@ public AnalyzedRestoreSnapshot analyze(RestoreSnapshot<Expression> restoreSnapsh
+                includeGlobalSettings = true;
+            }
+            case METADATA -> {
+                includeCustomMetadata = true;
+                includeGlobalSettings = true;
+                globalSettings.add(AnalyzerSettings.CUSTOM_ANALYSIS_SETTINGS_PREFIX);
+            }
+            case TABLE -> {
+                includeTables = true;
+                var exprAnalyzerWithFieldsAsString = new ExpressionAnalyzer(
+                    txnCtx,
+                    nodeCtx,
+                    paramTypeHints,
+                    FieldProvider.FIELDS_AS_LITERAL,
+                    null
+                );
+                tables = Lists2.map(
+                    restoreSnapshot.tables(),
+                    (table) -> table.map(x -> exprAnalyzerWithFieldsAsString.convert(x, exprCtx))
+                );
+            }
+            case CUSTOM -> {
+                for (String type_name : restoreSnapshot.types()) {
+                    type_name = type_name.toUpperCase(Locale.ENGLISH);
+                    if (type_name.equals("TABLES")) {
+                        includeTables = true;
+                    } else if (type_name.equals("ANALYZERS")) {
+                        // custom analyzers are stored inside persistent cluster settings
+                        globalSettings.add(AnalyzerSettings.CUSTOM_ANALYSIS_SETTINGS_PREFIX);
+                        includeGlobalSettings = true;
+                    } else {
+                        var custom_type = METADATA_CUSTOM_TYPE_MAP.get(type_name);
+                        if (custom_type == null) {
+                            throw new IllegalArgumentException("Unknown metadata type '" + type_name + "'");
+                        }
+                        includeCustomMetadata = true;
+                        customMetadataTypes.add(custom_type);
+                    }
+                }
+            }
+            default -> { }       // make checkstyle happy
```

Could we throw an assertion error here?